### PR TITLE
enabling 21.04

### PR DESCRIPTION
**Pull request by bam241**
testing 21.04

### CODE REVIEW

File: .github/workflows/21.04.yml [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+env:
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+jobs:
+  main:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        cubit: [2021.5]
+        os: [ubuntu-20.04]
```
Comment (gonuke) [inline .github/workflows/21.04.yml]:
```suggestion
        os: [ubuntu-21.04]
```

**Review by gonuke — CHANGES_REQUESTED** [review]
Make sure we name the job correctly

**Comment by bam241** [comment]
This was a test :)

B. Mouginot

File: .github/workflows/21.04.yml (hunk 1 above)
Comment (shimwell) [reply .github/workflows/21.04.yml]: Just going to commit this suggestion to retrigger the CI and generate another assert for testing locally

**Comment by shimwell** [comment]
Just for anyone reading this. This PR is just building the plugin using docker images instead of the base github action unix. The reson we are testing this is to produce the cubit plugin with different glibc versions. It appears that the version of glibc used when using the base github action unix to build is a bit older than versions available on Ubuntu 20.04, Ubuntu 21.04 and Debian. Glbc is tricky to upgrade so it is nice to have the plugin built with a range of versions

File: .github/workflows/21.04.yml [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+      - master
+  release:
+    types: # This configuration does not affect the page_build event above
+      - created
+env:
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+jobs:
+  main:
+    runs-on: ${{ matrix.os }}
```
Comment (shimwell) [inline .github/workflows/21.04.yml]:
```suggestion
    runs-on: ubuntu-latest
```

File: .github/workflows/21.04.yml [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+    strategy:
+      matrix:
+        cubit: [2021.5]
+        os: [ubuntu-21.04]
+   
+
+    name: 'Cubit Svalinn Plugin ${{ matrix.cubit }} Build for ${{ matrix.os }}'
+
+    container:
+      image: ubuntu:21.04
```
Comment (Shimwell) [inline .github/workflows/21.04.yml:32]:
```suggestion
      image: ${{ matrix.os }}
```

File: .github/workflows/21.04.yml (hunk 1 above)
Comment (bam241) [reply .github/workflows/21.04.yml]: we need 20.04 GitHub action doesn't support 21.04 just yet...
this is running in a docker...

File: .github/workflows/21.04.yml [hunk 4]
```diff
@@ -0,0 +1,153 @@
+name: Unix
```
Comment (shimwell) [inline .github/workflows/21.04.yml]:
```suggestion
name: Multiple-Unix
```

File: .github/workflows/21.04.yml [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+env:
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+jobs:
+  main:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        cubit: [2021.5]
+        os: [ubuntu-20.04]
```
Comment (shimwell) [inline .github/workflows/21.04.yml]:
```suggestion
      matrix:
        cubit: [2021.5]
        include:
          - image: ubuntu:21.04
            os: ubuntu-21.04
          - image: ubuntu:20.04
            os: ubuntu-20.04
          - image: debian:10.10
            os: debian-10.10
```

File: .github/workflows/21.04.yml [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+   
+
+    name: 'Cubit Svalinn Plugin ${{ matrix.cubit }} Build for ${{ matrix.os }}'
+
+    container:
+      image: ubuntu:21.04
```
Comment (shimwell) [inline .github/workflows/21.04.yml]:
```suggestion
      image:  ${{ matrix.image }}
```

**Comment by shimwell** [comment]
Ah sorry I have broken it again, I think the updating git trick that works for ubuntu doesn't quite work for debian

**Comment by shimwell** [comment]
Looks like my attempt to add debian was a bit ambitious, the ```unix_share_build.sh``` script contains commands that install ```lsb-core``` and ```ubuntu_version``` that are not valid in debian 10.10

**Comment by shimwell** [comment]
I've started working on the debian build over here https://github.com/Shimwell/Cubit-plugin/pull/2 sorry for all the mess

**Comment by shimwell** [comment]
I think we can close this test PR now. Feel free to reopen if still needed

**Pull request closed without merging**